packages/ra-core/src/dataProvider/useGetOne.ts

```diff
@@ -42,7 +42,7 @@ export const useGetOne = <RecordType extends Record = Record>(
 ): UseGetOneHookValue<RecordType> => {
     const dataProvider = useDataProvider();
     return useQuery<RecordType, unknown, RecordType>(
-        [resource, 'getOne', id],
+        [resource, 'getOne', String(id)],
```

Comment:
Out of curiosity, why do we need to stringify it?

Reply:
Because the react-query cache is type-sensitive, and sometimes the id comes as a string (e.g. when read from the URL in a Show view), sometimes as a number (e.g. when read from a Record in useGetList response).

But since you're asking the question, I realize I need to add that comment to the code.